cmd/autoscaler/main.go

```diff
@@ -282,6 +282,10 @@ func informerFilter(listOptions *kmeta.ListOptions) {
 }
 
 func exit(log *zap.SugaredLogger, err error, wrapStrs ...string) {
+	if err == nil {
+		os.Exit(0)
+	}
+
 	for _, str := range wrapStrs {
 		err = errors.Wrap(err, str)
 	}
```

Comment:
This function will still not exit if `errors.IsNoPrint(err)` is true. Also, the `if err != nil` check below can be removed (since we check it above)

Reply:
solved